freezer/freezer_url_test.go

```diff
@@ -122,3 +177,39 @@ func TestFreezerSource(t *testing.T) {
 	}
 
 }
+
+type mockStore struct {
+	url string
```

Comment:
You can just embed the interface and then you can avoid having to specify all the methods below. The behaviour will be the same as it will panic when it tries to use a method when the store is not set. Not really important though...
```suggestion
	straw.StreamStore
	url string
```

Reply:
nice approach. I'll push a change (it needs more than just this 1-liner to work)